src/Elasticsearch/Util/FieldDatatypeTrait.php

```diff
@@ -69,7 +69,7 @@ private function getNestedFieldPath(string $resourceClass, string $property): ?s
             ) {
                 $nestedPath = $this->getNestedFieldPath($nextResourceClass, implode('.', $properties));
 
-                return null === $nestedPath ? $nestedPath : "$currentProperty.$nestedPath";
+                return null === $nestedPath ? $currentProperty : "$currentProperty.$nestedPath";
```

Comment:
@jonnyeom I have to revert this to make https://github.com/api-platform/core/blob/main/features/elasticsearch/match_filter.feature#L120 green any idea? more at https://github.com/api-platform/core/pull/5823

Reply:
interesting.
Let me take a deeper look again.

Thanks for heads up!